tests/ai/src/index.ts

```diff
@@ -0,0 +1,166 @@
+  });
+
+  const prompt = new ChatPrompt({
+    instructions: 'You are a friendly assistant who talks like a pirate',
+    model,
+  });
+
+  const response = await prompt.send(activity.text);
+  if (response.content) {
+    const activity = new MessageActivity(response.content).addAiGenerated();
+    await send(activity);
+    // Ahoy, matey! 🏴‍☠️ How be ye doin' this fine day on th' high seas? What can this ol’ salty sea dog help ye with? 🚢☠️
+  }
+});
+// :snippet-end:
+
+// Handle "<supported-command> <query>" message
+app.on('message', async ({ send, activity, next }) => {
+  if (activity.text.toLowerCase().startsWith('docs ')) {
+    await handleDocumentationSearch(
+      model,
+      {
+        ...activity,
+        text: activity.text.slice(5),
+      },
+      send
+    );
+    return;
+  }
+
+  const commandAndQuery = [pokemonCommand, weatherCommand, feedbackLoopCommand, ragCommand]
+    .map((command) => command(activity.text))
+    .find(Boolean);
+  if (!commandAndQuery) {
+    await next();
+    return;
+  }
+  const { commandName, query, handler } = commandAndQuery;
+  if (!handler) {
+    console.warn(`Command ${commandName} does not have a supplied handler`);
+  } else {
+    await handler(
+      model,
+      {
+        ...activity,
+        text: query,
+      },
+      send
+    );
+  }
+});
+
+// Handle messages that start with stream <query>
+// :snippet-start: streaming-chat
+app.on('message', async ({ stream, send, activity, next }) => {
+  // :remove-start:
+  const commandAndQuery = streamCommand(activity.text);
+  if (!commandAndQuery) {
+    await next();
+    return;
+  }
+  const { query } = commandAndQuery;
+  // :remove-end:
+  // const query = activity.text;
+
+  const prompt = new ChatPrompt({
+    instructions: 'You are a friendly assistant who responds in terse language',
+    model,
+  });
+
+  // Notice that we don't `send` the final response back, but
+  // `stream` the chunks as they come in
+  const response = await prompt.send(query, {
+    onChunk: (chunk) => {
+      stream.emit(chunk);
+    },
+  });
+
+  if (activity.conversation.isGroup) {
+    // If the conversation is a group chat, we need to send the final response
+    // back to the group chat
+    const activity = new MessageActivity(response.content).addAiGenerated();
+    await send(activity);
+  } else {
+    // We wrap the final response with an AI Generated indicator
+    stream.emit(new MessageActivity().addAiGenerated());
+  }
+});
+// :snippet-end:
+
+// Fall through conversation handler
+app.on('message', async ({ send, activity }) => {
+  await handleStatefulConversation(model, activity, send);
+});
+
+// :snippet-start: feedback-loop-handler
+app.on('message.submit.feedback', async ({ activity }) => {
+  const { reaction, feedback: feedbackJson } = activity.value.actionValue;
+  if (activity.replyToId == null) {
+    console.warn(`No replyToId found for messageId ${activity.id}`);
```

Comment:
Should use the `ConsoleLogger` for these.

Reply:
Oh good call